pkg/plugins/plugins.go

```diff
@@ -0,0 +1,63 @@
+package plugins
+
+// OpenShiftAzureConfig is a new, Red Hat owned type defined in an external project.  It
+// represents a superset of the configuration required for all steps in the plugin process *after*
+// the manifest is validated.
+//
+// OpenShiftAzureConfig is a versioned type.  It is expected that the configuration will change
+// between releases and may require migration steps in the future.  The storage of this type
+// is opaque to the plugins, they only require that it be passed in.  However, it must be retained
+// to allow it to be lifecycled from version to version.
+type OpenShiftAzureConfig struct{}
+
+// ManagedOpenShiftCluster is the external schema that is passed to the RP.  It is defined in
+// acs-engine.  ManagedOpenShiftCluster will reuse many of the lower level objects shared by
+// ManagedCluster but is separated in order to keep the relationship with AKS clear.
+type ManagedOpenShiftCluster struct{}
+
+// Validator interface is the first plugin in the chain.  It is responsible for validation of
+// both ManagedOpenShiftCluster.  ManagedOpenShiftCluster enters this process in a versioned state.
+// It is validated both according to the version and against the previous state of the world.
+// Versioned validation should occur first.  The manifest, if acceptable, may then be converted
+// into the internal representation and validated against the previous state of the world.
```

Comment:
> internal representation

Are there 2 types then?  Do both need to be mentioned explicitly in this file?

Reply:
yes, internal and external.  Still the debate of what internal is, I feel like we've settled on external being managed in this repo and separate from the acs-engine external types.